datafusion/catalog/src/table.rs

```diff
@@ -299,6 +317,68 @@ pub trait TableProvider: Debug + Sync + Send {
     }
 }
 
+#[derive(Debug, Clone, Default)]
+pub struct ScanArgs {
+    preferred_ordering: Option<Vec<SortExpr>>,
```

Comment:
I'm adding `preferred_ordering` here even though it's unused for now, if CI (or a reviewer) complains I'll remove it